### ISSUE TITLE
Create layout for tx list

### ISSUE THREAD

**Issue by elgatovital** [gnosis/safe-android#571]

**Comment by elgatovital** [comment]
screen: https://zpl.io/bzZ8zz8

**Comment by biafra23** [comment]
https://zpl.io/bzZ8zz8 seems to be the iOS screen. I cannot find the Android version in Zeplin. All the  transactions screens seem to be missing.

**Comment by biafra23** [comment]
The link to the Android screen is: https://app.zeplin.io/project/5e9efc30726e106958cc43de/screen/5eb1234bf944bdb12a3fd89c

**Comment by biafra23** [comment]
Does it make sense to have these three types of list items?
- payment
- settings_change
- function_call

![image.png](https://images.zenhubusercontent.com/5e97a1c48301b54c5cd938c1/445d04ad-11b0-4059-af15-95a5ded00fd7)

**Comment by rmeissner** [comment]
I would probably call them a little different, but in general yes.

`[...]_transfer` - this is a asset transfer (can be erc20 or erc721, so not always a payment)
`[...]_custom_transaction` - this is a transaction where the user entered the code
`[...]_settings_change` - this changes the settings of the safe, so the to address is the safe itself

And there is this special one
`[...]_safe_creation` - For this is not actually transaction from the backend, it needs to be added by the client at the end of the list.

The names are inspired from how we call them on the safe-react interface

**Comment by biafra23** [comment]
As per our (Vitaly, Jose and me) discussion today I will make the following separate layout files:

`item_pending_transaction.xml` (used for any type of tx: transfer, custom_transaction & settings_change while pending)
`item_historical_transaction.xml` (used for transfers and custom_transactions when canceled, failed or succeeded)
`item_historical_settings_change.xml` (used for settings_change when canceled, failed or succeeded)

**Comment by elgatovital** [comment]
it would suggest having same prefix for tx layout items: item_tx or item_transaction. This way everything will be bundled together in res folder.
item_transaction_queued.xml
item_transaction_transfer.xml
item_transaction_internal.xml or item_transaction_settings.xml

**Comment by jpalvarezl** [comment]
> And there is this special one
> `[...]_safe_creation` - For this is not actually transaction from the backend, it needs to be added by the client at the end of the list.

@rmeissner pagination will be an issue for this. Is this requirement set on stone? I think a more suitable place for this view would be in the "Safe Settings > Advanced" . I am not sure what the purpose of that button is, but it seems like it could fit. Otherwhise we could just add another named "Safe settings > Safe creation parameters"

**Comment by rmeissner** [comment]
So while there might be a more appropriate place (that is a UX question ;) ) it would be interesting to know how pagination influences this? Backend tells you if it is the last page so you could just insert a row at the end. The challenge would be some details (date etc, as they require another request), but a simple "Safe created" row that show more info on click should be simple, right?

I think this entry was initially created so that the list is not empty right after creation. 

TL; DR: yes pagination is a must. How the creation transaction will be returned will be decided next week.

**Comment by jpalvarezl** [comment]
Ah ok, well yeah I didn't think of that. I was just putting myself in the scenario where someone may have tones of Txs and then this item is pushed down into the very bottom, but I have no idea if there are people with that many TXs. Besides, I think I do see the value on not having an empty TX list from the get go. It's a bit of an intuitive onboarding where we can show off the TX details and hint at deeper navigation destinations. 

Maybe having both what I mentioned and this could also make sense. Well, we will see next week either way.

**Comment by elgatovital** [comment]
> I think a more suitable place for this view would be in the "Safe Settings > Advanced" 
Safe Settings > Advanced is about current settings for safe => currently enabled modules and fallback handler. @jpalvarezl  
![image.png](https://images.zenhubusercontent.com/5ccabf56658809463cff37ed/2aa0c601-e1c6-41a5-b896-999203d8897c)
Safe creation will show data as it was during deployment.

**Comment by biafra23** [comment]
> So while there might be a more appropriate place (that is a UX question ;) ) it would be interesting to know how pagination influences this? Backend tells you if it is the last page so you could just insert a row at the end. The challenge would be some details (date etc, as they require another request), but a simple "Safe created" row that show more info on click should be simple, right?
> 
> I think this entry was initially created so that the list is not empty right after creation.
> 
> TL; DR: yes pagination is a must. How the creation transaction will be returned will be decided next week.

Just to clarify: The safe created item should always be shown as the last item of the list. Regardless whether the user is on the last page?

**Comment by rmeissner** [comment]
No, only on the last page

**Comment by biafra23** [comment]
I don't think it is useful to only have one pending layout file. The contract changes layouts for pending are also quite different. I would propose the follwowing two dimensional matrix of layouts.

1st dimension: queued | final
2nd dimension: transfer&custom_tx | contract_setting | change_mastercopy

That would give us 6 item layouts:
`item_tx_queued_transfer.xml`
![image](https://user-images.githubusercontent.com/246473/81413368-96faa100-9145-11ea-8d24-9d1e71d2c596.png)

`item_tx_queued_settings_change.xml` (design by me)
![fake_item_tx_queued_settings_change](https://user-images.githubusercontent.com/246473/81413744-3a4bb600-9146-11ea-8f19-d0682652d244.png)

`item_tx_queued_change_mastercopy.xml`
![fake_item_tx_queued_master_copy](https://user-images.githubusercontent.com/246473/81413903-81d24200-9146-11ea-8f1d-608699d5340d.png)



`item_tx_final_transfer.xml`
`item_tx_final_settings_change.xml`
`item_tx_final_change_mastercopy.xml`

**Comment by biafra23** [comment]
For external reference in the wiki:

Latest design for History:

![image](https://user-images.githubusercontent.com/246473/83238718-21f41780-a197-11ea-8674-ed8e94a03fb4.png)

And queued:

![image](https://user-images.githubusercontent.com/246473/83238760-33d5ba80-a197-11ea-9a99-49b854da6de9.png)
